### ISSUE TITLE
SC2016 shouldn't happen for single quoted variable INSIDE double-quoted string 

### ISSUE THREAD

**Issue by dmlane** [koalaman/shellcheck#3092]
#### For bugs
- SC2016:
- version: 0.10.0
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
var1="abc"
var2="$(a_function '${var1}')"
echo "$var2"
```

#### Here's what shellcheck currently says:
Line 3:
var2="$(a_function '${var1}')"
                   ^-- [SC2016](https://www.shellcheck.net/wiki/SC2016) (info): Expressions don't expand in single quotes, use double quotes for that.

#### Here's what I wanted or expected to see:
It shouldn't be flagged as an error. Bash *does* expand ${var1} because the outer delimiters are double quotes.

**Comment by dmlane** [comment]
I'm sorry - I thought my code was working, but I did some more testing and I'm wrong. I'll close the issue.
